### ISSUE TITLE
Add support for reporting row and column headers in contenteditable HTML elements.

### ISSUE THREAD

**Issue by msftedad** [nvaccess/nvda#14113]
### Steps to reproduce:
**Repro Steps:**
1. Open the URL: https://ckeditor.com/ckeditor-4/demo/
1. Turn on NVDA.
2. Turn on NVDA Focus mode.
3. Add a new table and add column headers for the table
4. Now using ctrl + alt + arrow keys to navigate from one cell to another
5. Verify whether NVDA is announcing column headers or not.

Code pen for minimal sample: https://codepen.io/testaccount01/pen/rNvxdwJ

### Actual behavior (Speech Viewer):-
For 1st table: -
row 2
column 1
row 1  a
row 2
row 3
column 2
column 2  b
column 1
row 3
column 2

For 2nd Table (Speech Viewer) :-
column 2  H2
column 1  H1
row 2  R1
column 2
column 3
row 1  H3
row 2
row 3
column 2
row 1  H3

NVDA is not announcing row or column headers.
-->

### Expected behavior:
NVDA should announce row or column headers when we navigate from one cell to another cell. When it is going to the new cell it should announce the column headers of the new cell.

### NVDA logs, crash dumps and other attachments:
[[2137516_NVDA is not announcing headers as headers in table in CKEditor demo (1).zip](https://github.com/nvaccess/nvda/files/9504016/2137516_NVDA.is.not.announcing.headers.as.headers.in.table.in.CKEditor.demo.1.zip)](https://microsoftapc-my.sharepoint.com/:u:/g/personal/v-deenthakur_microsoft_com/EaKv3yxacN1MmmAJ7Wui7DEBIrBRE87w_QnNbzrbUP3kGQ?e=AddeMk)

#### NVDA installed/portable/running from source:
Installed

#### NVDA version:
NVDA Version 2022.2

#### Windows version:
Windows 11 Version 10.0.22000 Build 22000

#### Name and version of other software in use when reproducing the issue:
Microsoft Edge Version 105.0.1343.27 (Official build) (64-bit)

#### Other information about your system:
N/A

### Other questions
N/A

#### Does the issue still occur after restarting your computer?
Yes

#### Have you tried any other versions of NVDA? If so, please report their behaviors.
No

#### If NVDA add-ons are disabled, is your problem still occurring?
Yes

#### Does the issue still occur after you run the COM Registration Fixing Tool in NVDA's tools menu?
Yes

**Comment by seanbudd** [comment]
We are unable to process this issue as it stands.

- Please fill out the issue template fully.
- To be able to reproduce this issue, we need a code sample. Provide a minimal HTML sample to reproduce this using [codepen](https://codepen.io/).
- Can you confirm what you mean by "NVDA scan mode", do you mean "Browse Mode"?

**Comment by feerrenrut** [comment]
This is a missing feature in NVDA and would be considered an important and valuable addition.

**Comment by Adriani90** [comment]
Note that for example Microsoft 365 Word online is basically built on such contentEditable structures, so solving thin in NVDA would also improve probably alot the experience in Word Online.

**Comment by michaelDCurran** [comment]
For anyone implementing this, take a look at the `_getControlFieldForObject` method on class `CompoundTextInfo` in `compoundDocuments.py`.
It should be a simple as filling in `table-rowheadertext` and `table-columnheadertext` attributes using values of `obj.rowHeaderText` and `obj.columnHeaderText` respectively.
Here is a very simple [test_contenteditable_table.html.txt](https://github.com/nvaccess/nvda/files/13700459/test_contenteditable_table.html.txt). Make sure to switch to focus mode, and use control+alt+arrow keys to move around the table.
It is expected that NvDA would report the appropriate day column header and time row header when moving between the cells in this calendar.
